### ISSUE TITLE
Refactor sqlite insert data from file.

### ISSUE THREAD

**Issue by henrykironde** [weecology/retriever#940]
`insert_data_from_file()` in the sqlite engine uses `open` `with open(filename, 'r') as data_file:`.
This has been a problem in the past.
We can move the block of procedure to be handled `gen_from_source(data_source)`.

_gen_from_source_ returns the rows in a list, we could yield the generator to the CHUNK_SIZE for each insert

```
            try:
                bulk_insert_statement = self.get_bulk_insert_statement()
                line_endings = set(['\n', '\r', '\r\n'])
                with open(filename, 'r') as data_file:
                    data_chunk = data_file.readlines(CHUNK_SIZE)
                    data_chunk = [line.rstrip('\r\n') for line in data_chunk if line not in line_endings]
                    del(data_chunk[:self.table.header_rows])
                    while data_chunk:
                        data_chunk_split = [row.split(self.table.delimiter)
                                            for row in data_chunk]
                        self.cursor.executemany(bulk_insert_statement, data_chunk_split)
                        data_chunk = data_file.readlines(CHUNK_SIZE)
```

**Comment by shreyneil** [comment]
@henrykironde how to go about this one, i am unable to get a clear idea?

**Comment by henrykironde** [comment]
We decided to stick to the old way. This was solved in #912 by reusing the old method.
